src/Analyser/NodeScopeResolver.php

```diff
@@ -1909,6 +1910,18 @@ function (MutatingScope $scope) use ($expr, $nodeCallback, $context): Expression
 			$throwPoints = [];
 			if ($expr->name instanceof Expr) {
 				$nameType = $scope->getType($expr->name);
+				if (
```

Comment:
This looks like it would not support a union of object types. 

Could you test that?

Reply:
The `TypeWithClassName` check wasn't excluding ClosureType, so I went back this way.
```
$nameType->isObject()->yes()
					&& !$nameType instanceof ClosureType
```

Is there a way to do this check without an `instanceof *Type` ? I know PHPStan codebase got rid of them.

Reply:
Would Type->getClassReflections() work?

Reply:
I found a way